pkg/e2etemplates/aws_operator_chart_values.go

```diff
@@ -20,37 +22,127 @@ const AWSOperatorChartValues = `Installation:
-                token: ${GUEST_AWS_SESSION_TOKEN}
+                id: {{ .Secret.AWSOperator.SecretYaml.Service.AWS.AccessKey.ID }}
+                secret: {{ .Secret.AWSOperator.SecretYaml.Service.AWS.AccessKey.Secret }}
+                token: {{ .Secret.AWSOperator.SecretYaml.Service.AWS.AccessKey.Token }}
               hostaccesskey:
-                id: ${HOST_AWS_ACCESS_KEY_ID}
-                secret: ${HOST_AWS_SECRET_ACCESS_KEY}
-                token: ${HOST_AWS_SESSION_TOKEN}
+                id: {{ .Secret.AWSOperator.SecretYaml.Service.AWS.HostAccessKey.ID }}
+                secret: {{ .Secret.AWSOperator.SecretYaml.Service.AWS.HostAccessKey.Secret }}
+                token: {{ .Secret.AWSOperator.SecretYaml.Service.AWS.HostAccessKey.Token }}
 
       Registry:
         PullSecret:
-          DockerConfigJSON: "{\"auths\":{\"quay.io\":{\"auth\":\"${REGISTRY_PULL_SECRET}\"}}}"
+          DockerConfigJSON: "{\"auths\":{\"quay.io\":{\"auth\":\"{{ .RegistryPullSecret }}\"}}}"
     Security:
       RestrictAccess:
         Enabled: false
 `
+
+type AWSOperatorChart struct {
+	Guest    AWSOperatorChartGuest
+	Provider AWSOperatorChartProvider
+	Secret   AWSOperatorChartSecret
+
+	RegistryPullSecret string
+}
+
+func (a AWSOperatorChart) Validate() error {
+	if a.Provider.AWS.Region == "" {
+		return microerror.Maskf(invalidDataError, "%T.Provider.AWS.Region must not be empty", a)
+	}
+	if a.Secret.AWSOperator.IDRSAPub == "" {
+		return microerror.Maskf(invalidDataError, "%T.Secret.AWSOperator.IDRSAPub must not be empty", a)
+	}
+	if a.Secret.AWSOperator.SecretYaml.Service.AWS.AccessKey.ID == "" {
+		return microerror.Maskf(invalidDataError, "%T.Secret.AWSOperator.SecretYaml.Service.AWS.AccessKey.ID must not be empty", a)
+	}
+	if a.Secret.AWSOperator.SecretYaml.Service.AWS.AccessKey.Secret == "" {
+		return microerror.Maskf(invalidDataError, "%T.Secret.AWSOperator.SecretYaml.Service.AWS.AccessKey.Secret must not be empty", a)
+	}
+	if a.Secret.AWSOperator.SecretYaml.Service.AWS.AccessKey.Token == "" {
+		return microerror.Maskf(invalidDataError, "%T.Secret.AWSOperator.SecretYaml.Service.AWS.AccessKey.Token must not be empty", a)
+	}
+	if a.Secret.AWSOperator.SecretYaml.Service.AWS.HostAccessKey.ID == "" {
+		return microerror.Maskf(invalidDataError, "%T.Secret.AWSOperator.SecretYaml.Service.AWS.HostAccessKey.ID must not be empty", a)
+	}
+	if a.Secret.AWSOperator.SecretYaml.Service.AWS.HostAccessKey.Secret == "" {
+		return microerror.Maskf(invalidDataError, "%T.Secret.AWSOperator.SecretYaml.Service.AWS.HostAccessKey.Secret must not be empty", a)
+	}
+	if a.Secret.AWSOperator.SecretYaml.Service.AWS.HostAccessKey.Token == "" {
+		return microerror.Maskf(invalidDataError, "%T.Secret.AWSOperator.SecretYaml.Service.AWS.HostAccessKey.Token must not be empty", a)
+	}
+
+	if a.RegistryPullSecret == "" {
+		return microerror.Maskf(invalidDataError, "%T.RegistryPullSecret must not be empty", a)
+	}
+
+	return nil
+}
+
```

Comment:
Types. I followed what @MarcelMue introduced wherever I could. We should align things down the road.